src/extension/wps/src/main/java/au/org/emii/gogoduck/worker/GoGoDuckModule.java

```diff
@@ -18,30 +32,107 @@
     private static final String TIME_FIELD = "time";
     private static final String URL_FIELD = "file_url";
 
-    protected String profile = null;
-    protected GoGoDuckSubsetParameters subset = null;
-    protected UserLog userLog = null;
-    protected IndexReader indexReader = null;
+    @Autowired
+    ServletContext context;
+
+    private String profile = null;
+    private GoGoDuckSubsetParameters subset = null;
+    private UserLog userLog = null;
+    private IndexReader indexReader = null;
 
-    public GoGoDuckModule() {}
+    private Boolean postProcessingRequired = null;
+    private Boolean timeUnlimited = null;
+    private GoGoDuckConfig goGoDuckConfig;
 
-    public void init(String profile, IndexReader indexReader, String subset, UserLog userLog) {
+    public GoGoDuckModule() { }
+
+    public GoGoDuckModule(String profile, IndexReader indexReader, String subset, UserLog userLog, GoGoDuckConfig goGoDuckConfig) {
+        init(profile, indexReader, subset, userLog, goGoDuckConfig);
+    }
+
+    public void init(String profile, IndexReader indexReader, String subset, UserLog userLog, GoGoDuckConfig goGoDuckConfig) {
         this.profile = profile;
         this.indexReader = indexReader;
         this.subset = new GoGoDuckSubsetParameters(subset);
         this.userLog = userLog;
+        this.goGoDuckConfig = goGoDuckConfig;
     }
 
     public URIList getUriList() throws GoGoDuckException {
```

Comment:
GoGoDuckModule doesn't have to own the GoGoDuckSubset object it can be instantiated in GoGoduck and passed to both the GoGoDuckModule and the UrlIndexReader.  - they are really doing separate things and they don't need to be in the same class anymore.  Having said that, its just cleaning up the code (the way I think it should be cleaned up) and is optional if you want to focus on getting the core processing to work again.